### PR TITLE
[shopsys] removed read model package

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Read model package is obsolete thanks to the frontend API and new JS Storefront implementation. This PR removes unused code with no longer necessary package.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Model/Product/ProductOnCurrentDomainFacade.php [hunk 1]
```diff
@@ -51,154 +49,27 @@ public function getVisibleProductById(int $productId): Product
...
+        string $searchText = '',
     ): ProductFilterCountData {
         $productsQueryBuilder = $this->productRepository->getListableInCategoryQueryBuilder(
             $this->domain->getId(),
             $this->currentCustomerUser->getPricingGroup(),
             $this->categoryRepository->getById($categoryId),
         );
 
+        // database filtering does not support limit by search
+
```
Comment (TomasLudvik) [inline packages/framework/src/Model/Product/ProductOnCurrentDomainFacade.php]: This makes me think - is the implementation for Postgres still needed? Should we remove it too? I vote for removing.

File: packages/read-model/README.md [hunk 2]
```diff
@@ -1,63 +0,0 @@
-# Shopsys Read Model
-
-[![Downloads](https://img.shields.io/packagist/dt/shopsys/read-model.svg)](https://packagist.org/packages/shopsys/read-model)
```
Comment (TomasLudvik) [inline packages/read-model/README.md:3]: Should not we keep it and say it is no longer in use? Or we just abandon it?

File: packages/framework/tests/Unit/Component/Image/DirectoryStructureCreatorTest.php [hunk 3]
```diff
@@ -37,8 +37,8 @@ public function testMakeImageDirectories(): void
         $filesystemMock
             ->method('createDirectory')
             ->withConsecutive(
-                ['imageDir/entityName1/'],
-                ['imageDir/entityName2/type/'],
+                ['imageDir/entityName1'],
+                ['imageDir/entityName2/type'],
```
Comment (TomasLudvik) [inline packages/framework/tests/Unit/Component/Image/DirectoryStructureCreatorTest.php]: Why is this needed during this change?

**Review by TomasLudvik — CHANGES_REQUESTED** [review]

File: packages/framework/tests/Unit/Component/Image/DirectoryStructureCreatorTest.php (hunk 3 above)
Comment (grossmannmartin) [reply packages/framework/tests/Unit/Component/Image/DirectoryStructureCreatorTest.php]: This PR is based on the currently in progress `rv-images` branch. This is merely a fix for the base branch, so I can be sure the tests are passing. 
After the base branch is merged, I don't expect this change to be part of this PR.

File: packages/read-model/README.md (hunk 2 above)
Comment (grossmannmartin) [reply packages/read-model/README.md:3]: The package may still be valid for the 12.0 LTS version. 
By removing it from the monorepo we just stop the auto-releasing along with the rest of the Shopsys Platform. 

We may also delete the 14.0 branch in the shopsys/read-model package.

File: packages/framework/src/Model/Product/ProductOnCurrentDomainFacade.php (hunk 1 above)
Comment (grossmannmartin) [reply packages/framework/src/Model/Product/ProductOnCurrentDomainFacade.php]: removed :)

File: docs/model/front-end-product-filtering.md [hunk 4]
```diff
@@ -8,14 +8,11 @@ These two pages are represented by `ProductController` and `SearchController`, w
 -   `getPaginatedProductsInCategory()` to obtain filtered products in category
 -   `getPaginatedProductsForSearch()` to obtain filtered products from search results
 
-Currently, there are two implementations of `ProductOnCurrentDomainFacadeInterface`:
+Currently, there are single implementation of `ProductOnCurrentDomainFacadeInterface`:
```
Comment (TomasLudvik) [inline docs/model/front-end-product-filtering.md]: there is

**Review by TomasLudvik — APPROVED** [review]

File: docs/model/front-end-product-filtering.md (hunk 4 above)
Comment (grossmannmartin) [reply docs/model/front-end-product-filtering.md]: fixed
